libs/ui/src/ui_strings/audio_context.tsx

```diff
@@ -3,28 +3,133 @@ import React from 'react';
 import { Optional } from '@votingworks/basics';
 
 import { UiStringsReactQueryApi } from '../hooks/ui_strings_api';
+import {
+  DEFAULT_PLAYBACK_RATE,
+  MAX_PLAYBACK_RATE,
+  MIN_PLAYBACK_RATE,
+  PLAYBACK_RATE_INCREMENT_AMOUNT,
+} from './audio_playback_rate';
+import {
+  GAIN_INCREMENT_AMOUNT_DB,
+  DEFAULT_GAIN_DB,
+  MAX_GAIN_DB,
+  MIN_GAIN_DB,
+} from './audio_volume';
 
-export interface AudioContextInterface {
-  // TODO(kofi): Flesh out.
+export interface UiStringsAudioContextInterface {
+  api: UiStringsReactQueryApi;
+  decreasePlaybackRate: () => void;
+  decreaseVolume: () => void;
+  gainDb: number;
+  increasePlaybackRate: () => void;
+  increaseVolume: () => void;
+  isEnabled: boolean;
+  playbackRate: number;
+  reset: () => void;
+  setIsEnabled: (enabled: boolean) => void;
+  togglePause: () => void;
+  webAudioContext?: AudioContext;
 }
 
-const AudioContext =
-  React.createContext<Optional<AudioContextInterface>>(undefined);
+export const UiStringsAudioContext =
+  React.createContext<Optional<UiStringsAudioContextInterface>>(undefined);
 
-export function useAudioContext(): Optional<AudioContextInterface> {
-  return React.useContext(AudioContext);
+export function useAudioContext(): Optional<UiStringsAudioContextInterface> {
+  return React.useContext(UiStringsAudioContext);
 }
 
-export interface AudioContextProviderProps {
-  // eslint-disable-next-line react/no-unused-prop-types
+export interface UiStringsAudioContextProviderProps {
   api: UiStringsReactQueryApi;
   children: React.ReactNode;
 }
 
-export function AudioContextProvider(
-  props: AudioContextProviderProps
+export function UiStringsAudioContextProvider(
+  props: UiStringsAudioContextProviderProps
 ): JSX.Element {
-  const { children } = props;
+  const { api, children } = props;
+  const [isEnabled, setIsEnabled] = React.useState(false);
+  const [isPaused, setIsPaused] = React.useState(true);
+  const [playbackRate, setPlaybackRate] = React.useState<number>(
+    DEFAULT_PLAYBACK_RATE
+  );
+  const [gainDb, setGainDb] = React.useState<number>(DEFAULT_GAIN_DB);
+
+  const webAudioContextRef = React.useRef(
+    window.AudioContext ? new AudioContext() : undefined
+  );
+
+  const reset = React.useCallback(() => {
+    setGainDb(DEFAULT_GAIN_DB);
+    setPlaybackRate(DEFAULT_PLAYBACK_RATE);
+    setIsPaused(false);
+  }, []);
+
+  React.useEffect(() => {
+    if (isEnabled) {
+      reset();
+    } else {
+      setIsPaused(true);
```

Comment:
Just checking my understanding: Are we pausing here because disabling the audio player necessitates pausing any in-flight audio?